cadence/contracts/bridge/FlowEVMBridgeHandlerInterfaces.cdc

```diff
@@ -0,0 +1,132 @@
+///
+access(all) contract FlowEVMBridgeHandlerInterfaces {
+
+    /******************
+        Entitlements
+    *******************/
+
+    /// Entitlement related to administrative setters
+    access(all) entitlement Admin
+    /// Entitlement related to minting handled assets
+    access(all) entitlement Mint
+
+    /*************
+        Events
+    **************/
+    
+    /// Event emitted when a handler is enabled between a Cadence type and an EVM address
+    access(all) event HandlerEnabled(handlerType: Type, targetType: Type, targetEVMAddress: EVM.EVMAddress)
+
+    /****************
+        Constructs
+    *****************/
+    
+    /// Non-privileged interface for querying handler information
+    ///
+    access(all) resource interface HandlerInfo {
+        /// Returns whether the Handler is enabled
+        access(all)	view fun isEnabled(): Bool
+        /// Returns the Cadence type handled by the Handler, nil if not set
+        access(all) view fun getTargetType(): Type?
+        /// Returns the EVM address handled by the Handler, nil if not set
+        access(all) view fun getTargetEVMAddress(): EVM.EVMAddress?
+    }
+
+    /// Administrative interface for Handler configuration
+    ///
+    access(all) resource interface HandlerAdmin : HandlerInfo {
+        /// Sets the target Cadence Type handled by this resource
+        access(Admin) fun setTargetType(_ type: Type) {
+            pre {
+                self.getTargetType() == nil: "Target Type has already been set"
+            }
+            post {
+                self.getTargetType()! == type: "Problem setting target type"
+            }
+        }
+        /// Sets the target EVM address handled by this resource
+        access(Admin) fun setTargetEVMAddress(_ address: EVM.EVMAddress) {
+            pre {
+                self.getTargetEVMAddress() == nil: "Target EVM address has already been set"
+            }
+            post {
+                self.getTargetEVMAddress()!.bytes == address!.bytes: "Problem setting target EVM address"
+            }
+        }
+        /// Enables the Handler to fulfill bridge requests for the configured targets
+        access(Admin) fun enableBridging() {
+            pre {
+                self.getTargetType() != nil && self.getTargetEVMAddress() != nil:
+                    "Cannot enable before setting bridge targets"
+                !self.isEnabled(): "Handler already enabled"
+            }
+            post {
+                self.isEnabled(): "Problem enabling Handler"
+                emit HandlerEnabled(
+                    handlerType: self.getType(),
+                    targetType: self.getTargetType()!,
+                    targetEVMAddress: self.getTargetEVMAddress()!
+                )
+            }
+        }
+    }
+
+    /// Minter interface for configurations requiring the minting of Cadence fungible tokens
+    ///
+    access(all) resource interface TokenMinter {
```

Comment:
So in a contract that wants to be supported here, all they would have to do is have their minter resource implement this interface?

If we expect this implementation to happen as part of the cadence 1.0 upgrade, we need to deploy this interface on all the networks before then so the migrations work properly

Reply:
Yeah, I didn't see a way around it if we want to store minters with specific interface guarantees. My thought was that we would deploy this contract after the migration and any implementers would then update their contracts, but that feels cumbersome...

Maybe we could deploy this contract with just the `TokenMinter` interface before the migration so that the contract could be migrated and included in dependent contract updates. And ofc stage this contract which shouldn't be a problem since all of its dependencies will also be migrated. Think that would work?

Reply:
yep, that sounds good to me